# HG changeset patch
# User Marco Caliari <marco.caliari@univr.it>
# Date 1524725231 -7200
#      Thu Apr 26 08:47:11 2018 +0200
# Branch stable
# Node ID 60fd2d84c9a04be67c917bcb9b5a094ebe59e152
# Parent  159b359ccc13886065a24d02f8d031afc2758524
eigs.m: Fix failing BIST test (bug #53700).

* eigs.m: Use isreal() to determine whether NaN or NaN+1i*NaN should be the
expected value.

diff --git a/scripts/sparse/eigs.m b/scripts/sparse/eigs.m
--- a/scripts/sparse/eigs.m
+++ b/scripts/sparse/eigs.m
@@ -1425,17 +1425,21 @@ endfunction
 %!   assert (d(4), NaN +1i*NaN);
 %! endif
 %!testif HAVE_ARPACK
 %! A = magic (100) / 10 + eye (100);
 %! opts.v0 = (1:100)';
 %! opts.maxit = 10;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (A, 10, "sm", opts);
-%! assert (d(10), NaN+1i*NaN);
+%! if (isreal (d))
+%!   assert (d(10), NaN);
+%! else
+%!   assert (d(10), NaN +1i*NaN);
+%! endif
 %!testif HAVE_ARPACK
 %! A = toeplitz ([0, 1, zeros(1, 8)], [0, -1, zeros(1, 8)]);
 %! A = kron (A, eye (10)) + kron (eye (10), A);
 %! Afun = @(x) A * x;
 %! opts.v0 = (1:100)';
 %! opts.maxit = 5;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (Afun, 100, 4, "lm", opts);
